### ISSUE TITLE
❓ <title>only give normal_dir in Folder class

### ISSUE THREAD

**Issue by aikangjun** [open-edge-platform/anomalib#2949]
### Your Question

当我使用Folder实例化一个datamodule时，只传入了normal_dir,而未传入abnormal_dir和mask_dir。导出的模型在推理结果上差异非常大。代码如下:
train.py
`from anomalib.data import MVTecAD, Folder
from anomalib.deploy import ExportType
from anomalib.engine import Engine
from anomalib.models import Patchcore

# 2. Create a dataset
# MVTecAD is a popular dataset for anomaly detection
# datamodule = MVTecAD(
#     root=r"C:\Users\chenyinhui\PycharmProjects\dataset\MVTecAD",  # Path to download/store the dataset
#     category="carpet",  # MVTec category to use
#     train_batch_size=32,  # Number of images per training batch
#     eval_batch_size=32,  # Number of images per validation/test batch
#     num_workers=0
# )
datamodule = Folder(
    name="btad_01",
    normal_dir=r"C:\Users\chenyinhui\PycharmProjects\dataset\btad\BTech_Dataset_transformed\01\train\ok",
    # abnormal_dir=r"C:\Users\chenyinhui\PycharmProjects\dataset\btad\BTech_Dataset_transformed\01\test\ko",
    # mask_dir=r"C:\Users\chenyinhui\PycharmProjects\dataset\btad\BTech_Dataset_transformed\01\ground_truth\ko",
    train_batch_size=32,
    eval_batch_size=32,
    num_workers=0,
    seed=42
)

# 3. Initialize the model
# Patchcore is a good choice for beginners
model = Patchcore(
    num_neighbors=6,  # Override default model settings
    coreset_sampling_ratio=0.05  # 从训练样本中，选择多少比例的像素作为核心集
)

# 4. Create the training engine
engine = Engine(
    max_epochs=1,  # Override default trainer settings
)

# 5. Train the model
# This produces a lightning model (.ckpt)
engine.fit(datamodule=datamodule, model=model)

# 6. Export the model
# Different formats are available: Torch, OpenVINO, ONNX
engine.export(
    model=model,
    export_type=ExportType.TORCH,
)` 
inference.py:
`
from anomalib.models import Patchcore
from anomalib.engine import Engine
import torch
from anomalib.visualization.image.item_visualizer import visualize_image_item  # 可视化函数
import matplotlib.pyplot as plt

model = torch.load("results/Patchcore/btad_01/v3/weights/torch/model.pt")["model"]
model.to("cuda")
engine = Engine()
while True:
    data_path = input("image path or directory：")
    predictions = engine.predict(model,
                                 data_path=data_path)
    # 可视化：提取第一个（唯一）预测结果并生成图像 + 热力图叠加 + 分类标签
    if predictions and len(predictions) > 0:
        for batch in predictions:
            item = batch.items[0]  # ImageItem（单张图像时为 batch_size=1）
            # 生成可视化：图像 + 热力图叠加
            vis_result = visualize_image_item(
                item=item,
                overlay_fields=[("image", ["anomaly_map"])],
                text_config={"enable": False}
            )
            # 使用 Matplotlib 显示
            plt.figure(figsize=(8, 6))
            plt.imshow(vis_result)
            plt.axis("off")

            # 添加标题：分类类别和分数
            label_text = "Normal" if item.pred_label == 0 else "Anomalous"
            score_text = f"Predicted: {label_text} (Score: {item.pred_score:.3f})"
            plt.title(score_text, fontsize=14, color="red" if item.pred_label == 1 else "green")

            plt.tight_layout()
            plt.show()  # 显示图像
            print(f"预测分数: {item.pred_score:.3f}, 标签: {label_text}")
    else:
        print("预测失败：无结果返回，请检查路径和模型。")`
我仔细检查了PatchCore的逻辑，确实是在训练时才会修改memory_bank。那么按道理来说，结果不应该相差那么大才对。请帮助我解答一下疑问，谢谢！

### Forum Check

- [x] I have searched the discussions forum for an answer to my question.

**Comment by aikangjun** [comment]
我知道了，如果没有异常图片，那么验证集的F1-score就计算不出来。而在验证阶段会学习一个最佳阈值，分离正常和异常图片。如果阈值没有设置好，可视化的效果就会变差
